tests/e2e/testsuites/testsuites.go

```diff
@@ -365,8 +366,25 @@ func (t *TestPersistentVolumeClaim) ValidateProvisionedPersistentVolume() {
 				To(HaveLen(1))
 		}
 		if len(t.storageClass.AllowedTopologies) > 0 {
-			Expect(t.persistentVolume.Spec.NodeAffinity.Required.NodeSelectorTerms[0].MatchExpressions[0].Key).
-				To(Equal(t.storageClass.AllowedTopologies[0].MatchLabelExpressions[0].Key))
+			// Since we're chaging our topology key, assume we have the values below to compare:
+			// NodeSelectorTerms: [{[{topology.ebs.csi.aws.com/zone In [us-west-2a]} {topology.kubernetes.io/zone In [us-west-2a]}] []}]
+			// AllowedTopologies: [{[{topology.ebs.csi.aws.com/zone [us-west-2a us-west-2b us-west-2c]}]}]
+			// As you can see tests might fail depending on the ordering of the NodeSelectorTerms. That's why we're doing this "hack".
+			// This is a quick fix to unblock the PRs we have. We really need to improve this. TODO
+
+			keyFound := false
+			for _, v := range t.persistentVolume.Spec.NodeAffinity.Required.NodeSelectorTerms[0].MatchExpressions {
+				if v.Key == "topology.ebs.csi.aws.com/zone" {
+					keyFound = true
+					Expect(v.Key).To(Equal(t.storageClass.AllowedTopologies[0].MatchLabelExpressions[0].Key))
```

Comment:
This statement is pointless, it's a tautology, if you're in this if statement then this is already guaranteed to be true

Reply:
never mind, I confused storageClass allowedTopoloigies with the PV spec field? I'm a bit lost TBH,whether my SC has the old key or new key, the PV should have both old or new key right?

Reply:
Yeah PV has the both keys, but SC has one (in e2e tests it's always the oldest one, see above tests/e2e/driver/ebs_csi_driver.go). So we have to make sure we're comparing the correct keys. 

For example, this test fails when `t.persistentVolume.Spec.NodeAffinity.Required.NodeSelectorTerms[0].MatchExpressions[0].Key` is not `topology.ebs.csi.aws.com/zone`, which is possible because we also have `topology.kubernetes.io/zone` in the slice.